yt/frontends/ahf/io.py

```diff
@@ -67,15 +69,13 @@ def _identify_fields(self, data_file):
 
     # Helper methods
 
-    def _get_data_files(self, chunks, ptf):
+    @staticmethod
+    def _validate_particle_ptf(ptf):
```

Comment:
this function is now 100% `assert` statements, which means it can be treated as noop by linters/type-checkers (because `assert`s can be ignored at runtime). I suggest rewriting it with actual an `if/raise` clause. It's also possible to *leverage* the type checking system so that this function actually helps narrowing down the type of `ptf` where it's used, using `TypeGuard`. IMO this would also help reading the code (I had no idea what `ptf` stood like or what type of data was assumed until I researched it for the sake of this suggestion)

```suggestion
    def _validate_particle_ptf(ptf: Dict[str, Any]) -> TypeGuard[HalosPTF]:
        if list(ptf) != ["halos"]:
            raise ValueError(...)
        return True
```

with
```python
from typing import TypedDict, Any
class HalosPTF(TypedDict):
    halos: Any
```

(I'm using `Any` mostly because I'm not sure what type these values are)

Reply:
a bit of extra context -- I only added this function because this particular frontend had lumped in the validation of this dictionary to a `_get_data_files` function and I wanted to separate out the validation so that the call signature of `_get_data_files` matches the new base class function. So it seemed convenient to add this new function, but there are actually at least 3-4 other frontends that do this **exact** assert validation within the body of `_read_particle_fields` and `_read_particle_coords`, so I could simply do that here rather than adding the new function. I do think raising errors would indeed be better, but it may be simpler to refactor all the halo particle readers at once to raise errors instead of assert (in a separate PR). 

also happy to just use your suggestion :)

Reply:
It would probably be easier to refactor them all at once, and I don't want my suggestion to leak way out of scope here so it's probably best to preserve the similarity for now

Reply:
OK, that sounds good to me. I'll go ahead and move these asserts to the body of the functions then.

Reply:
done